src/topology/metric_space/isometry.lean

```diff
@@ -230,6 +212,55 @@ protected def to_homeomorph (h : α ≃ᵢ β) : α ≃ₜ β :=
   h.to_homeomorph.to_equiv = h.to_equiv :=
 rfl
 
+section normed_group
+
+variables {G : Type*} [normed_group G]
+
+/-- Addition `y ↦ y + x` as an `isometry`. -/
+protected def add_right (x : G) : G ≃ᵢ G :=
+{ isometry_to_fun := isometry_emetric_iff_metric.2 $ λ y z, dist_add_right _ _ _,
+  .. equiv.add_right x }
+
+@[simp] lemma add_right_to_equiv (x : G) :
+  (isometric.add_right x).to_equiv = equiv.add_right x := rfl
+
+@[simp] lemma add_right_apply (x y : G) : (isometric.add_right x : G → G) y = y + x := rfl
+
+@[simp] lemma add_right_symm (x : G) :
+  (isometric.add_right x).symm = isometric.add_right (-x) :=
+ext $ λ y, rfl
+
+/-- Addition `y ↦ x + y` as an `isometry`. -/
+protected def add_left (x : G) : G ≃ᵢ G :=
+{ isometry_to_fun := isometry_emetric_iff_metric.2 $ λ y z, dist_add_left _ _ _,
+  to_equiv := equiv.add_left x }
+
+@[simp] lemma add_left_to_equiv (x : G) :
+  (isometric.add_left x).to_equiv = equiv.add_left x := rfl
+
+@[simp] lemma coe_add_left (x : G) : ⇑(isometric.add_left x) = (+) x := rfl
```

Comment:
Should the lemmas be consistent between `add_right`, `add_left` and `neg`? E.g. should we have `add_left_apply` (and `neg_apply` and `coe_add_right`) as well?

Reply:
I added `@[simp] lemma coe_add_right`. My reason for adding `add_right_apply` (and not others) is that with `coe_add_left` you can rewrite right to left from `a + b` to `add_left a b` while `coe_add_right` won't let you do the same.